book/running-wallaroo/resilience.md

```diff
@@ -4,72 +4,110 @@ Wallaroo is designed with built-in resilience. However, since the operations inv
 
 Since recovery involves a rollback to the last successful checkpoint, any data that was processed _after_ that checkpoint will have to be resent by the sources.
 
 ## Recovery in Practice
 
 When a crashed worker is restarted, if it can find its resilience files in the path specified by `--resilience-dir`, it will automatically start the recovery process. The simplest way to do this is to rerun the worker using the same command it was originally run with.
 
-For example, if we were running the bundle Python example [word_count](/examples/python/word_count_with_dynamic_keys/README.md), this might look something like:
+For example, if we were running the bundled Python example [word_count_with_dynamic_keys](https://github.com/WallarooLabs/wallaroo/tree/master/examples/python/word_count_with_dynamic_keys/), this might look something like:
 
-1. Start a data receiver
+### Running the resilience example
 
-        data_receiver --listen 127.0.0.1 7000 | tee received.txt
- 
-2. Start initializer
+In order to run the example you will need Machida with resilience enabled, Giles Sender, Data Receiver, and the Cluster Shutdown tool. If you're using our Docker or Vagrant installation, these should already be set up.
 
-        machida --application-module word_count_with_dynamic_keys --in 127.0.0.1:7010 \
-        --out 127.0.0.1:7002 --metrics 127.0.0.1:5001 --control 127.0.0.1:6000 \
-        --data 127.0.0.1:6001 --name worker-name --external 127.0.0.1:5050 \
-        --cluster-initializer --ponythreads=1 --ponynoblock \
-        --resilience-dir /tmp/res-dir/wc \
-        | tee initializer.log
+If you built Wallaroo from source, please run the following commands:
 
-3. Start worker
+    cd ~/wallaroo-tutorial/wallaroo-{{ book.wallaroo_version }}/
+    make build-machida-all resilience=on
+    cp machida/build/machida bin/machida-resilience
 
-        machida --application-module word_count_with_dynamic_keys --in 127.0.0.1:7010 \
-        --out 127.0.0.1:7002 --metrics 127.0.0.1:5001 --join 127.0.0.1:6000 \
-        --my-data 127.0.0.1:6003 --my-control 127.0.0.1:6002 --name worker1 \
-        --external 127.0.0.1:5051 --ponythreads=1 --ponynoblock \
-        --resilience-dir /tmp/res-dir/wc \
-        | tee worker1.log
+If you haven't yet set up wallaroo, please visit our [setup](https://docs.wallaroolabs.com/book/getting-started/choosing-an-installation-option.html) instructions to get started.
 
-4. Start sender
+### Running the example
 
+The rest of the example should be run in the `examples/python/word_count_with_dynamic_keys` directory in the Wallaroo repository.
+You will need 4 terminals to run this example.
+
+1. Create the path where the workers will save their resilience snapshots.
+
+        mkdir -p /tmp/resilience-dir
+
+2. Terminal 1: Start a data receiver
```

Comment:
Should we be using `Shell` instead of `Terminal` to be consistent with the rest of the book?